Tools/ardupilotwaf/ardupilotwaf.py

```diff
@@ -204,6 +204,26 @@ def ap_find_tests(bld, use=[]):
             cxxflags=['-Wno-undef'],
         )
 
+_versions = []
+
+@conf
+def ap_version_append(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, os.environ.get(k, v))]
+
+@conf
+def ap_version_append_str(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, '"{}"'.format(os.environ.get(k, v)))]
+
+@conf
+def write_version_header(ctx):
```

Comment:
Or have a function in this file to create the taskgen. So that the header name uses are somewhat together.